app/models/split_registry_snapshot.rb

```diff
@@ -0,0 +1,37 @@
+class SplitRegistrySnapshot
+  include ActiveModel::Validations
+
+  validates :timestamp, presence: true
+  # We want to make sure the client is shipping us high-precision ISO
+  # timestamps so we choose to allow timestamps at either millisecond or second
+  # precision based on the W3C interpretation of iso8601 from this SO answer:
+  #
+  # https://stackoverflow.com/a/3143231
+  validates :timestamp, format: {
+    with: /\A\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d(.\d+)?([+-][0-2]\d:[0-5]\d|Z)\z/, allow_blank: true
+  }
+
+  def initialize(timestamp:)
```

Comment:
`as_of`? it lines up with the naming convention in `Split` and clarifies what the timestamp is for

Reply:
There still needs to be an initializer though, which probably needs this signature, right? .as_of could be added as a sugar class method tho? Or am I misinterpreting?

Reply:
sorry, my comment wasn't clear, i meant changing the arg name from `timestamp` to `as_of`.

Reply:
oh oh, i knew I was misunderstanding. :)